app/models/content_substitutor.rb

```diff
@@ -27,11 +29,19 @@ def confirmation_email_body
       service_name: service_name
     )
 
-    substitute_placeholder(
-      setting: setting,
-      placeholder: REFERENCE_NUMBER_PLACEHOLDER,
-      content: body_content
-    )
+    if payment_link_enabled
+      substitute_placeholder(
+        setting: setting,
+        placeholder: REFERENCE_PAYMENT_PLACEHOLDER,
+        content: reference_payment_body_content
+      )
+    else
+      substitute_placeholder(
+        setting: setting,
+        placeholder: REFERENCE_PAYMENT_PLACEHOLDER,
+        content: body_content
+      )
+    end
```

Comment:
```
      substitute_placeholder(
        setting: setting,
        placeholder: REFERENCE_PAYMENT_PLACEHOLDER,
        content: reference_payment_body_content
      )
```

maybe let the `reference_payment_body_content` method decide what the content should be?